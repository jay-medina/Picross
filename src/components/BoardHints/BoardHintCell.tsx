import React from 'react';

export interface BoardHintCellProps {
  hint: number;
}

const BoardHintCell: React.StatelessComponent<BoardHintCellProps> = ({ hint }) => (
  <div className="cell hint">
    {hint}
  </div>
);

export default BoardHintCell;
