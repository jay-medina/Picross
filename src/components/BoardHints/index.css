.boardHints {
  display:flex;
}

.boardHints.column {
  flex-direction: column;
}
