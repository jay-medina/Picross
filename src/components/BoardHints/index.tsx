import React from 'react';
import BoardHintCell from './BoardHintCell';
import './index.css';

export interface BoardHintsProps {
  direction: 'row' | 'column';
  hints: number[];
}

class BoardHints extends React.PureComponent<BoardHintsProps, {}> {
  render() {
    return (
      <div className="boardHints">
        {this.renderCells()}
      </div>
    );
  }
  private renderCells() {
    return this.props.hints.map(hint => <BoardHintCell hint={hint} />);
  }
}

export default BoardHints;
