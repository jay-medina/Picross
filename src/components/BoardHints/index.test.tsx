import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import BoardHints from './index';

describe('<BoardHints />', () => {
  let wrapper: ShallowWrapper;

  describe('when rendering a row of board hints', () => {
    beforeEach(() => {
      wrapper = shallow(
        <BoardHints direction="row" hints={[1, 2, 3]} />,
      );

    });

    it('renders the cells', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
