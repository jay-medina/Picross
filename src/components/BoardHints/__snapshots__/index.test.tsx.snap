// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BoardHints /> when rendering a column of board hints renders the cells 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BoardHints
    direction="column"
    hints={
      Array [
        1,
        2,
        3,
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <BoardHintCell
          hint={1}
        />,
        <BoardHintCell
          hint={2}
        />,
        <BoardHintCell
          hint={3}
        />,
      ],
      "className": "boardHints column",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "hint": 1,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "hint": 2,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "hint": 3,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <BoardHintCell
            hint={1}
          />,
          <BoardHintCell
            hint={2}
          />,
          <BoardHintCell
            hint={3}
          />,
        ],
        "className": "boardHints column",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "hint": 1,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "hint": 2,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "hint": 3,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<BoardHints /> when rendering a row of board hints renders the cells 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BoardHints
    direction="row"
    hints={
      Array [
        1,
        2,
        3,
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <BoardHintCell
          hint={1}
        />,
        <BoardHintCell
          hint={2}
        />,
        <BoardHintCell
          hint={3}
        />,
      ],
      "className": "boardHints",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "hint": 1,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "hint": 2,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "hint": 3,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <BoardHintCell
            hint={1}
          />,
          <BoardHintCell
            hint={2}
          />,
          <BoardHintCell
            hint={3}
          />,
        ],
        "className": "boardHints",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "hint": 1,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "hint": 2,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "hint": 3,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
